LoRaEngine/LoraKeysManagerFacade/LoRaADRFunction/LoRaADRRedisStore.cs

```diff
@@ -14,27 +14,36 @@ namespace LoraKeysManagerFacade
 
     public class LoRaADRRedisStore : LoRaADRStoreBase, ILoRaADRStore
     {
-        const string CacheToken = "ADR";
-        const string LockToken = "_lock";
+        const string CacheToken = ":ADR";
+        const string LockToken = ":lock";
         IDatabase redisCache;
 
         sealed class RedisLockWrapper : IDisposable
         {
+            private static readonly TimeSpan LockTimeout = TimeSpan.FromSeconds(10);
             private string lockKey;
             private string owner;
             private IDatabase redisCache;
             private bool ownsLock;
 
-            internal RedisLockWrapper(string devEUI, IDatabase redisCache, string owner = "_LoRaRedisStore")
+            internal RedisLockWrapper(string devEUI, IDatabase redisCache, string owner = ":LoRaRedisStore")
             {
                 this.lockKey = GetEntryKey(devEUI) + LockToken;
                 this.redisCache = redisCache;
                 this.owner = owner;
             }
 
-            internal bool TakeLock()
+            internal async Task<bool> TakeLockAsync()
             {
-                return this.ownsLock = this.redisCache.LockTake(this.lockKey, this.owner, TimeSpan.FromSeconds(10));
+                var start = DateTime.UtcNow;
+                while (!(this.ownsLock = await this.redisCache.LockTakeAsync(this.lockKey, this.owner, TimeSpan.FromSeconds(10))))
+                {
+                    if (DateTime.UtcNow - start > LockTimeout)
+                        break;
+                    await Task.Delay(100);
```

Comment:
Are these timings going to be fixed or could they maybe be declared as constants?

Reply:
The delay you mean? That's really just a loop delay. We can make the lock duration a constant. Not sure if you meant that.

Generally: I'm not 100% happy about the code at the moment. The problem is that we have 2 wrappers around redis cache, which is harder to maintain. That's also why we didn't see this locking issue when we fixed it in the first wrapper. Something we should cleanup if we get a chance.

Reply:
I second that, the Redis wrapper are confusing. We should revisit whenever possible.